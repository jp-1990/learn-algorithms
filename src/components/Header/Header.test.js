import React from "react"
import { render } from "@testing-library/react"
import renderer from "react-test-renderer"

import Header from "./Header"

// To update the snapshot to reflect valid changes run npm test -- -u
describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Header siteTitle="Gatsby Default Starter" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("has expected title", () => {
    // gets id from data id set on component
    const { getByTestId } = render(
      <Header siteTitle="Gatsby Default Starter" />
    )
    // assertion
    expect(getByTestId("header-title")).toHaveTextContent(
      "Gatsby Default Starter"
    )
  })
})
