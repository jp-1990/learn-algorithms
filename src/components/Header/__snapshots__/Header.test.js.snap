// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <div>
    <h1
      data-testid="header-title"
    >
      <a
        href="/"
      >
        Gatsby Default Starter
      </a>
    </h1>
  </div>
</header>
`;
